test/check/check.go

```diff
@@ -553,6 +591,51 @@ func checkForMutatingWebhookConfiguration(name string) error {
 	return nil
 }
 
+func checkForService(name string) error {
```

Comment:
in the following functions, it makes the monitoring mandatory isn't it ?
opposed to the fact that you allow CNAO to be deployed even if the monitoring has an error

Reply:
The tests checking monitoring components should be skipped if the CRDs are not deployed..
I do skip the test checking it specifically if the CRDs are not deployed, but you might be right in the case of [AllComponents](https://github.com/kubevirt/cluster-network-addons-operator/pull/928/files#diff-dd3d0d54317e754fd95de914de01a73b5578626b42e755fdad4998186d552600R89)... hmmm.. let me see if I can dynamically remove it

Reply:
Best lets inject an error that won't deploy monitoring and see what fails ? (unless you did that already)
it will what must be changed first

Reply:
I was thinking of wrapping `AllComponents` with `getAllComponents()` that will check and exclude monitoring if it is not deployed.. what do you think?

Reply:
So.. if we only keep the most updated provider, then I think we can keep `AllComponents`.
Since we always deploy prometheus, this the CRDs should always be deployed, and we should make sure that it acts as expected in our environment.
However, on production we don't want to force the user to deploy monitoring or CNAO will fail.